apps/emqx_ds_builtin_raft/src/emqx_ds_replication_layer_meta.erl

```diff
@@ -733,6 +750,32 @@ ensure_site() ->
             logger:error("Attempt to claim site with ID=~s failed: ~p", [Site, Reason])
     end.
 
+file_read_term(Filename) ->
+    %% NOTE
+    %% This mess is needed because `file:consult/1` trips over binaries encoded as
+    %% latin1, which 5.4.0 code could have produced with `io:format(FD, "~p.", [Site])`.
+    maybe
+        {ok, FD} ?= file:open(Filename, [read, {encoding, latin1}]),
+        {ok, Term, _} ?= io:read(FD, '', _Line = 1),
+        ok = file:close(FD),
+        {ok, Term}
+    else
+        {error, Reason} ->
+            {error, Reason};
+        {error, Reason, _} ->
+            {error, Reason};
+        {eof, _} ->
+            {error, empty}
+    end.
+
+migrate_site_id(Site) ->
+    case re:run(Site, "^[0-9A-Z]+$") of
```

Comment:
This looks probabilistic... Minor nit: hex cannot contain characters from `[G-Z]` range

Reply:
To some degree. Current implementation essentially only cares about them being "printable", in the sense that it's safe to feed them into `binary_to_atom/1` and `~s` format args.